### PR TITLE
Init entities and add migration

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- Ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/74978

## WHAT (optional)
- I create entities based on designed database, add relationships and migration

## HOW
- I create entity files in entities folder, add migration file to migration folder, add enums to constants.ts file

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/entities/class.entity.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { Student } from "./student.entity";
+import { Teacher } from "./teacher.entity";
+import { Teaching } from "./teaching.entity";
+
+@Entity()
+export class Class {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({ length: 255 })
```
Comment (huongnt-2545) [inline src/entities/class.entity.ts]: dùng constants đã define `255`

File: src/common/constants.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  STAFF = 1,
+  TEACHER = 2,
+  STUDENT = 3,
+}
+
+export enum EntityPropertyLength {
+  SMALL = 30,
+  MEDIUM = 50,
+  LARGE = 100,
+}
```
Comment (huongnt-2545) [inline src/common/constants.ts]: ko nên dùng enum ở đây e nha, đặt constants thông tg hỉ
vd:
MAX_LENGTH_50 = 50
...

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**